packages/api/es/indexer.js

```diff
@@ -20,8 +20,60 @@ const { Search } = require('./search');
 const Rule = require('../models/rules');
 const uniqBy = require('lodash.uniqby');
 
-async function indexLog(payloads, index = 'cumulus', type = 'logs') {
-  const esClient = await Search.es();
+/**
+ * Returns the collectionId used in elasticsearch
+ * which is a combination of collection name and version
+ *
+ * @param {string} name - collection name
+ * @param {string} version - collection version
+ * @returns {string} collectionId
+ */
+function constructCollectionId(name, version) {
+  return `${name}___${version}`;
+}
+
+/**
+ * Returns the name and version of a collection based on
+ * the collectionId used in elasticsearch indexing
+ *
+ * @param {string} collectionId - collectionId used in elasticsearch index
+ * @returns {Object} name and version as object
+ */
+function deconstructCollectionId(collectionId) {
+  const deconstructed = collectionId.split('___');
+  return {
+    name: deconstructed[0],
+    version: deconstructed[1]
+  };
+}
+
+/**
+ * Ensures that the exception is returned as a string
+ *
+ * @param {*} exception - the exception
+ * @returns {string} an stringified exception
+ */
+function prepareException(exception) {
+  if (typeof exception === 'object') {
+    return JSON.stringify(exception);
+  }
+  return exception;
+}
+
+/**
+ * Extracts granule info from a stepFunction message and indexes it to
+ * an ElasticSearch
+ *
+ * @param  {Object} esClient - ElasticSearch Connection object
+ * @param  {Array} payloads  - an array of log payloads
+ * @param  {string} index    - Elasticsearch index (default: cumulus)
+ * @param  {string} type     - Elasticsearch type (default: granule)
+ * @returns {Promise} Elasticsearch response
+ */
+async function indexLog(esClient, payloads, index = 'cumulus', type = 'logs') {
+  if (!esClient) {
```

Comment:
As far as I can tell, `indexLog` is only ever used here:  `return indexLog(undefined, logs.logEvents)`.  Is there an expectation that, at some point, some other code will be calling `indexLog` and passing it an esClient?

Reply:
so far it is only happening in the test file.

Reply:
It doesn't look like it's happening anywhere in the tests.

```
$ pwd
/Users/mhuffnagle/Source/cumulus/packages/api
$ grep -r 'indexLog' . | egrep -v '^./dist/'
./es/indexer.js:async function indexLog(esClient, payloads, index = 'cumulus', type = 'logs') {
./es/indexer.js:      return indexLog(undefined, logs.logEvents)
```